guacamole-ext/src/main/java/org/apache/guacamole/token/TokenFilter.java

```diff
@@ -156,19 +156,31 @@ public void setTokens(Map<String, String> tokens) {
         tokenValues.clear();
         tokenValues.putAll(tokens);
     }
-    
+
     /**
      * Filters the given string, replacing any tokens with their corresponding
-     * values.
+     * values. Handling of undefined tokens depends on the value given for the
+     * strict flag.
      *
      * @param input
      *     The string to filter.
      *
+     * @param strict
+     *     Whether to disallow tokens which lack values from existing in the
+     *     string. If true, an exception will be thrown if any tokens in the
+     *     string lack corresponding values. If false, tokens which lack values
+     *     will be interpreted as literals.
+     *
      * @return
      *     A copy of the input string, with any tokens replaced with their
      *     corresponding values.
+     *
+     * @throws GuacamoleTokenUndefinedException
+     *     If the strict flag is set to true and at least one token in the
+     *     given string has no corresponding value.
      */
-    public String filter(String input) {
+    private String filter(String input, boolean strict)
```

Comment:
Why `private`?  Is there any reason not to make this implementation available for calling outside this class?  Would it make sense to make this method public and deprecate the other version of the method, leaving it in (for now) for compatibility.

Reply:
(See https://github.com/apache/guacamole-client/pull/336#discussion_r771785428)

Reply:
My rationale here for keeping `filter(String, boolean)` private and instead relying on a dedicated `filterStrict()` is ease of verifiability. In cases where the behavior of `filterStrict()` is desirable, it's likely that calling `filter()` without the strict behavior would have security implications, and allowing that behavior to vary easily at runtime via a simple boolean parameter could be problematic.

Keeping `filter()` and `filterStrict()` extremely separate avoids that.